src/modules/RA/Picture/components/PictureInput.js

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import SideBySideImages from '@terralego/core/modules/Picture/compare/SideBySideImages';
+
+import {
+  useDataProvider,
+  useInput,
+  useLocale,
+  useTranslate,
+  ImageInput,
+  ImageField,
+
+} from 'react-admin';
+
+import Modal from '@material-ui/core/Modal';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+import {  makeStyles } from '@material-ui/core/styles';
+
+import { RES_VIEWPOINT } from '../../ra-modules';
+import PictureGallery from './PictureGallery';
+
+const useStyles = makeStyles({
+  picture: {
+    position: 'relative',
+    float: 'right',
+    width: '50%',
+    '& .previews div': {
+      width: '100%',
+      height: 'auto',
+    },
+    '& .previews img': {
+      width: '100%',
+      height: 'auto',
+      maxHeight: 'inherit',
+    },
+    '& .refusal': {
+      position: 'absolute',
+      bottom: '1em',
+      left: 0,
+      right: 0,
+      padding: '2em',
+      fontSize: '1.5em',
+      margin: '0 -8px 0 8px',
+      color: 'orange',
+      backgroundColor: 'rgba(23, 23, 23, 0.43)',
+    },
+  },
+  modal: {
+    top: '45%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    position: 'absolute',
+    height: '70%',
+    width: '80%',
+    backgroundColor: 'white',
+    padding: '2em',
+    '& .actions': {
+      paddingBottom: '2em',
+      display: 'flex',
+      justifyContent: 'space-between',
+    },
+    display: 'flex',
+    flexDirection: 'row-reverse',
+  },
+  compareContainer: {
+    height: '70%',
+    width: '80%',
+    position: 'absolute',
+    left: '110px',
+    top: '10px',
+  },
+  galleryContainer: {
+    height: '100%',
+    width: '130px',
+  },
+  compareTitle: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  picturesDates: {
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+});
+
+const PictureInput = props => {
+  const {
+    record: {
+      viewpoint,
+      id,
+      state,
+      properties: { refusal_message: refusalMessage } = {},
+    } = {},
+  } = props;
+
+  const { input: { value: pic } } = useInput(props);
+  const [showModal, setShowModal] = useState(false);
+  const [picturesList, setPicturesList] = useState([]);
+  const [selectedIndex, setSelectedIndex] = useState(null);
+
+  const classes = useStyles();
+  const dataProvider = useDataProvider();
+  const translate = useTranslate();
+  const locale = useLocale();
+
+  const toLocaleDateString = useCallback(dateString => {
```

Comment:
You can put that function in utils/date.js

Reply:
Done.